src/ModelObjects/component.rs

```diff
@@ -75,6 +75,12 @@ impl Component {
             panic!("Unable to retrieve location based on id: {}", name)
         }
     }
+
+    /// Checks if a location exists in the component, by checking the given location `id` match any location in the component
+    pub fn location_exists(self, id: &str) -> bool {
+        self.get_locations().iter().any(|x| x.id.as_str() == id)
+    }
```

Comment:
unused?

Reply:
This implementation is not used, so we have removed it now :) @t-lohse